controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -788,7 +789,9 @@ public CompletableFuture<CreateStreamStatus.Status> createStream(String scope, S
                                                                      long createTimestamp, long requestId) {
         log.debug(requestId, "createStream with resource called.");
         OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
-        if (config.getRetentionPolicy() == null && this.globalRetentionPolicy.get()) {
+        if (!config.getRetentionPolicy().getRetentionType().equals(RetentionType.NONE)
```

Comment:
This check is redundant. None is by definition not null.

Reply:
Modified as per new changes